parmed/structure.py

```diff
@@ -2118,43 +2118,76 @@ def omm_add_constraints(self, system, constraints, rigidWater):
             If True, water bonds are constrained regardless of whether
             constrains is None
         """
+
         if constraints is None and not rigidWater: return
         if constraints not in (None, app.HBonds, app.AllBonds, app.HAngles):
             raise ValueError("Unrecognized constraints option (%s)" %
                              constraints)
+
         length_conv = u.angstrom.conversion_factor_to(u.nanometer)
-        # Rigid water only
-        if constraints is None:
-            is_water = _settler(self)
+
+        constraint_bond_set = set()
+        constraint_angle_set = set()
+        is_water = _settler(self)
+
+        if constraints == app.AllBonds or constraints == app.HAngles:
```

Comment:
`app.AllBonds` and `app.HAngles` are better thought of as enums (or singletons).  Compare using `is`, not `==`